progress-bar.js

```diff
@@ -0,0 +1,93 @@
+(function(){
+    if (!document.addEventListener ||
```

Comment:
This prevents the code from running on older browsers which don't support the features we need.